### PR TITLE
Response ruby-enumerable-diagnostic - Larson

### PR DESCRIPTION
**Pull request by rlarsonil71**
Comfort: 4
Clarity: 4

I used my class notes, Enumerable trainings and the ruby-docs for Array, Range, Hash, Enumerable and File to help with this diagnostic.

### CODE REVIEW

File: lib/diagnostic.rb [hunk 1]
```diff
@@ -7,36 +7,36 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.max { |a, b| a.length <=> b.length }
```
Comment (jordanallain) [inline lib/diagnostic.rb:10]: 👍  Here's another way:

`list.max_by(&:length)`

File: lib/diagnostic.rb [hunk 2]
```diff
@@ -46,6 +46,11 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    # Read each line of file
+    File.open(filename) do |file|
+      file.each_with_index do |line|
+        line.chomp
+      end
+    end
```
Comment (jordanallain) [inline lib/diagnostic.rb:54]: You got really close here Ron, here's a way to do it:

```ruby
File.open(filename).reduce(String.new) { |a, e|
    a << e.chomp + ' ' }.strip
```

**Comment by jordanallain** [comment]
5/6

Awesome work Ron!

🇯🇴 🇯🇴 🇯🇴 🇯🇴 🇯🇴

**Comment by rlarsonil71** [comment]
Thanks for the feedback Jordan!

--Ron

**Pull request closed without merging**